### PR TITLE
1-D depth encoder + Unity car simulation with I-maze

### PR DESCRIPTION
**Pull request by chetan51**

### CODE REVIEW

**Comment by chetan51** [comment]
@subutai 
cc: @scottpurdy @BoltzmannBrain 

I think you only really need to review the Python code. As for the Unity stuff, just open the project in Unity and run it if you want to check that it works.

File: sensorimotor/sensorimotor/encoders/one_d_depth.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+from nupic.encoders.scalar import ScalarEncoder
+
+
+
+class OneDDepthEncoder(Encoder):
+  """
+  Given an array of numbers, each representing distance to the closest object,
+  returns an SDR representation of that depth data.
+
+  TODO: Add details
```
Comment (scottpurdy) [inline sensorimotor/sensorimotor/encoders/one_d_depth.py]: update?

File: vehicle-control/agent/run.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+
+
+# complete uniform
+# positions = [i*20 for i in range(36)]
+
+# forward uniform
+positions = [i*10 for i in range(18, 54)]
+
+run(positions)
```
Comment (scottpurdy) [inline vehicle-control/agent/run.py]: might put these lines in an `if __name__ == "__main__"` block

File: vehicle-control/agent/run.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import numpy
```
Comment (scottpurdy) [inline vehicle-control/agent/run.py]: needs copyright headers

**Comment by scottpurdy** [comment]
Looks nice and clean to me. Left a few comments. And the unit tests looked a little light but I think @subutai probably doesn't want us spending too much time on unit tests so up to you

**Comment by BoltzmannBrain** [comment]
It'd be better to have assertions in the tests, but, as Scott said, extensive testing shouldn't slow you down. :+1:

**Comment by cogmission** [comment]
Isn't 1-D Depth theoretically an oxymoron? :-P

**Comment by chetan51** [comment]
@cogmission 1-D scan of a depth sensor :)

**Comment by cogmission** [comment]
:)

**Comment by cogmission** [comment]
Is this related to the experiment you and Yuwei were doing with the robot and the layer stabilization?

**Comment by chetan51** [comment]
@cogmission This is related to the new goal-directed behavior research we're currently conducting at Numenta.

**Comment by cogmission** [comment]
Very cool!

**Pull request merged**